### PR TITLE
chore: ecto 3 compatibility

### PR DESCRIPTION
**Pull request by Miradorn**
Update Ecto dep(s)

Some methods are in ecto, others in ecto_sql, so I just depend on both.
Also the `__meta__.source` field no longer includes a tuple `{prefix, source}` but only `source`.

### CODE REVIEW

**Comment by smeevil** [comment]
Hello, @Miradorn! This is your first Pull Request that will be reviewed by [Ebert](https://ebertapp.io/), an automatic Code Review service. It will leave comments on this diff with potential issues and style violations found in the code as you push new commits. You can also see all the issues found on this Pull Request on its [review page](https://ebertapp.io/github/smeevil/ecto_translate/pulls/6). Please check our [documentation](https://docs.ebertapp.io) for more information.

**Comment by smeevil** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 3 possible new issues (including those that may have been commented here).

You can see more details about this review at https://ebertapp.io/github/smeevil/ecto_translate/pulls/6.

File: lib/ecto_translate.ex [hunk 1]
```diff
@@ -67,79 +67,112 @@ defmodule EctoTranslate do
...
+          ids = Enum.map(models, fn model -> model.id end)
+          %{__meta__: %{source: {_, translatable_type}}} = Enum.at(models, 0)
+
+          translations =
+            EctoTranslate
+            |> where(translatable_type: ^translatable_type, locale: ^locale)
+            |> where([t], t.translatable_id in ^ids)
+            |> unquote(repo).all
+
+          # FIXME this might not be the most optimized way to do this, but for now, it works :)
```
Comment (smeevil) [inline lib/ecto_translate.ex:161]: FIXME found

File: lib/ecto_translate.ex [hunk 2]
```diff
@@ -67,79 +67,112 @@ defmodule EctoTranslate do
...
+          Enum.map(models, fn model ->
+            attributes =
+              Enum.filter(translations, fn translation ->
+                translation.translatable_id == model.id
+              end)
+              |> Enum.map(fn record ->
+                {String.to_atom(Map.get(record, :field)), Map.get(record, :content)}
+              end)
+              |> Enum.into(%{})
```
Comment (smeevil) [inline lib/ecto_translate.ex:170]: `Enum.into/3` is more efficient than `Enum.map/2 |> Enum.into/2`

File: lib/ecto_translate.ex [hunk 3]
```diff
@@ -67,79 +67,112 @@ defmodule EctoTranslate do
...
+            |> where(
+              translatable_type: ^translatable_type,
+              translatable_id: ^translatable_id,
+              locale: ^locale
+            )
+            |> unquote(repo).all
+            |> Enum.map(fn record ->
+              {String.to_atom(Map.get(record, :field)), Map.get(record, :content)}
+            end)
+            |> Enum.into(%{})
```
Comment (smeevil) [inline lib/ecto_translate.ex:143]: `Enum.into/3` is more efficient than `Enum.map/2 |> Enum.into/2`

**Comment by smeevil** [comment]
Ebert has finished reviewing this Pull Request and has found:

* 3 possible new issues (including those that may have been commented here).
* 1 fixed issue! 🎉

You can see more details about this review at https://ebertapp.io/github/smeevil/ecto_translate/pulls/6.

**Comment by smeevil** [comment]
Hi ,

Sorry for the late reply, been kinda busy lately :)

I've updated the semaphore config so that the build runs again.
There is a small issue with running the tests now :

```
    ** (EXIT) exited in: GenServer.call(Ecto.Repo.Registry, {:associate, #PID<0.200.0>, {Ecto.Adapters.Postgres, %{cache: #Reference<0.3928171066.4016439301.92972>, opts: [timeout: 15000, pool_size: 10, pool: DBConnection.Ownership], pid: #PID<0.201.0>, sql: Ecto.Adapters.Postgres.Connection, telemetry: {:debug, [], [:ecto_translate, :repo, :query]}}}}, 5000)
        ** (EXIT) no process: the process is not alive or there's no process currently associated with the given name, possibly because its application isn't started
```
Could you take a look at this ?

**Comment by Miradorn** [comment]
Hey, no problem, thanks for getting back at all!
I will look into it, we hav ebeen using my fork for a quite a while in production without errors, so at last it shouldn't be a problem outside of tests 🙃

**Comment by Miradorn** [comment]
Think I found the error, will open new PR in a second

**Pull request merged**